HIP/hip-367.md

```diff
@@ -0,0 +1,238 @@
+---
+hip:367
+title: HIP Unlimited token associations per Account
+author: Anirudh Ghanta <anirudh.ghanta@hedera.com>
+working-group: Richard Bair<richard.bair@hedera.com>, Jasper Potts<jasper.potts@hedera.com>, Michael Tinker<michael.tinker@hedera.com>, Anirudh Ghanta<anirudh.ghanta@hedera.com>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+created: 2022-02-17
+discussions-to: [Discussion thread](https://github.com/hashgraph/hedera-improvement-proposal/discussions/380)
+updated: 2202-03-11
+---
+
+## Abstract
+Permits every account to hold an unlimited number of token associations. Retains the current pricing for each token association. Modifies the number of token associations returned in the `getAccountInfo` and `getAccountBalance` query. Changes the criteria for `cryptoDelete` transaction to successfully mark the account as `deleted`
+
+## Motivation
+The current Services API permits the user to have at most 1,000 token associations per account.
+A user can work around this limitation by creating multiple accounts, each of which is limited to
+1,000 token associations. For some use cases, this 1,000 token association limit is frustrating.
+Due to the current implementation, it makes state proofs large and expensive. In addition, the
+current implementation imposes excessive costs as the number of token associations per account
+grows.
+
+Finally, returning large numbers of token associations in the free queries can lead to excessively long query response times.
+
+## User stories
+
+- As a user I should be able to associate to any number of tokens without any limitations under one account instead of dispersing them across multiple accounts.
+- As a user I should be able to pay for as many token associations upfront as I want to, using the automatic association slots 
+- As a primary creator of an NFT collection, I should be able to mint a large number of NFT serials in a single collection
+- As a secondary NFT marketplace dApp, I should be able to create many thousands of collections (created by multiple primary creators of the collections), such that each collection can hold a large number of NFTs
+- As an avid collector of NFT, I want to manage my entire collection within the same account.
+
+## Specification
+Currently, we have a limit of 1000 token associations per account put in place which is dictated by the bootstrap property `tokens.maxPerAccount`.
+This is enforced everytime we associate token/tokens to an account. 
+
+When using the HederaTokenStore :
+```java
```

Comment:
Imo HIP's don't require code snippets; in part because the codebase is changing so often---for example, this excerpt no longer exists (as you know 😁).